src/test/java/com/avioconsulting/mule/connector/vault/provider/internal/connection/provider/ProxyConnectionTestCase.java

```diff
@@ -0,0 +1,169 @@
+                ).respond(
+                response()
+                        .withStatusCode(200)
+                        .withHeader("Content-Type", "application/json")
+                        .withBody("{\"lease_id\":\"\",\"renewable\":false,\"data\":null,\"auth\":{\"client_token\":\"MOCK_TOKEN\",\"accessor\":\"18bb8f89-826a-56ee-c65b-1736dc5ea27d\",\"policies\":[\"web\",\"stage\"],\"lease_duration\":3600,\"renewable\":true}}")
+        );
+        mockClient
+                .withSecure(true)
+                .when(
+                        request()
+                                .withMethod("GET")
+                                .withPath("/v1/secret/data/test/mysecret")
+                                .withHeader("X-Vault-Token", "MOCK_TOKEN")
+                                .withHeader("Host", "myvault.com:8200")
+                                .withHeader("X-Vault-Token", "MOCK_TOKEN")
+                                .withHeader("Proxy-Connection")
+                                .withHeader("Proxy-Authorization", "Basic dGVzdDpwYXNzd29yZA==")
+                ).respond(
+                response()
+                        .withStatusCode(200)
+                        .withHeader("Content-Type", "application/json")
+                        .withBody("{\"request_id\": \"69411f4b-fb02-171a-f66c-485f106e7f5c\",\"lease_id\": \"\",\"renewable\": false,\"lease_duration\": 0,\"data\": {\"data\": {\"att1\": \"test_value1\",\"att2\": \"test_value2\"},\"metadata\": {\"created_time\": \"2019-04-24T23:03:18.63231658Z\",\"deletion_time\": \"\",\"destroyed\": false,\"version\": 1}},\"wrap_info\": null,\"warnings\": null,\"auth\": null}")
+
+        );
+        mockClient
+                .withSecure(true)
+                .when(
+                        request()
+                                .withMethod("POST")
+                                .withPath("/v1/auth/aws/login")
+                                .withHeader("Host", "myvault.com:8200")
+                                .withHeader("Proxy-Connection")
+                ).respond(
+                response()
+                        .withStatusCode(200)
+                        .withHeader("Content-Type", "application/json")
+                        .withBody("{\"lease_id\":\"\",\"renewable\":false,\"data\":null,\"auth\":{\"client_token\":\"MOCK_TOKEN\",\"accessor\":\"18bb8f89-826a-56ee-c65b-1736dc5ea27d\",\"policies\":[\"web\",\"stage\"],\"lease_duration\":3600,\"renewable\":true}}")
+        );
+        mockClient
+                .withSecure(true)
+                .when(
+                        request()
+                                .withMethod("GET")
+                                .withPath("/v1/secret/data/test/mysecret")
+                                .withHeader("X-Vault-Token", "MOCK_TOKEN")
+                                .withHeader("Host", "myvault.com:8200")
+                                .withHeader("X-Vault-Token", "MOCK_TOKEN")
```

Comment:
Why do we need the vault token header twice?

Reply:
We don't. It looks like I accidentally copied that line.

Reply:
I removed the extra header line in the latest commit to the branch